### PR TITLE
Update to 17.31.28

### PR DESCRIPTION
**Pull request by mlehtima**
Add patch to fix build with latest libxml2.

### CODE REVIEW

File: rpm/libzypp.spec [hunk 1]
```diff
@@ -32,6 +35,9 @@ BuildRequires:  libsolv-tools >= 0.7.15
 BuildRequires:  pkgconfig(sigc++-2.0)
 BuildRequires:  glib2-devel
 BuildRequires:  pkgconfig(popt)
+BuildRequires:  pkgconfig(protobuf)
+BuildRequires:  pkgconfig(protobuf-lite)
+BuildRequires:  pkgconfig(readline)
```
Comment (saukko) [inline rpm/libzypp.spec:39]: Is this mandatory here? We have dropped readline by purpose to reduce dependencies from our core image.

**Comment by pvuorela** [comment]
Build seems to fail on xml reader type conversion. Perhaps this actually needs the libxml2 update?

File: rpm/libzypp.spec (hunk 1 above)
Comment (pvuorela) [reply rpm/libzypp.spec:39]: Grepping readline it looks like used only in zypp-tui and that having a static library. Wasn't able to confirm due to build failure, but perhaps that doesn't end up as dependency on the resulting main package(?)

Comment (mlehtima) [reply rpm/libzypp.spec:39]: It does not end up as dependency of the main package which gets installed on device, only devel package will depend on it

Comment (pvuorela) [reply rpm/libzypp.spec:39]: Sounds good to me then.

**Comment by pvuorela** [comment]
The current master has only a couple of commits more, but one is fixing build with libxml with #ifdef on version. Could bump the checkout that far and hopefully have this compiling with either current or updated ?

**Comment by mlehtima** [comment]
Indeed there is now 17.31.28 release in upstream which fixes build with both old and new libxml2 versions which makes life easier for us. I will update the PR to use that.

**Review by pvuorela — APPROVED** [review]

**Pull request merged**